### DISCUSSION TITLE
Scope/Dome driver.

### DISCUSSION THREAD

**Discussion by hbtorana** [Stellarium/stellarium#1790] [Ideas]
Hi,
Has anyone added support for G-code/3d printer driver boards... (as Dome/Scope/other controller?)

All the required hardware + additional inputs to control almost any cartesian coordinate style device.

Scopes, domes, anything really.. 

I've used a bog standard 2012 gen6 printer board/gcode to control servo steering on a robomower.

PWM Heaters, temp probes, stepper control, endstops, autohome, distance probes, doors, filament sensors.

Serial, bluetooth, wifi, 32bit boards/arm/arduino.  
Hardware and code readily available and maintained.

Thoughts?

**Comment by gzotti** [comment]
If you have created some gear and want to enable others to build your system, you can contribute a chapter with a detailed description to the user guide, and an optional plugin for your controller. If this description is too long, we could also simply mention it in the User Guide, and you keep your project separate (github or elsewhere).
